src/LLVM_Runtime_Linker.cpp

```diff
@@ -648,6 +649,10 @@ std::unique_ptr<llvm::Module> get_initial_module_for_target(Target t, llvm::LLVM
                     modules.push_back(get_initmod_posix_threads(c, bits_64, debug));
                 }
                 modules.push_back(get_initmod_posix_get_symbol(c, bits_64, debug));
+                //This is just to verify the hexagon_dma mock tests 
```

Comment:
I think instead of doing this, you should do:

- Rename `halide_hexagon_allocate_from_l2_pool` to `halide_locked_cache_malloc`, and `halide_hexagon_free_from_l2_pool` to `halide_locked_cache_free`
- Add a new initmod with a default locked cache allocator that just calls halide_malloc/halide_free
- For non-Hexagon targets, use default locked cache allocator
- For Hexagon targets, use the real hexagon cache allocator

Reply:
API names changed. 
Not using default locked cache allocator as in mock tests we are not using Locked Cache at all .